Sources_v3/ChatClient.swift

```diff
@@ -164,7 +164,7 @@ public class Client<ExtraData: ExtraDataTypes> {
     @Atomic private var connectionIdWaiters: [(String?) -> Void] = []
     
     /// The token of the current user. If the current user is anonymous, the token is `nil`.
-    @Atomic private var currentToken: Token?
+    @Atomic var currentToken: Token?
```

Comment:
I don't like making the `currentToken: Token` even internal as far as the only one who needs the access is **CurrentUserControllerGeneric**. 

I'd use a closure-based approach here but wanna discuss if it's not an overhead:
```swift

struct Getter<T> {
    private let closure: () -> T
    
    init(closure: @escaping () -> T) {
        self.closure = closure
    }
    
    var value: T {
        closure()
    }
}

struct Setter<T> {
    private let closure: (T) -> Void
    
    init(closure: @escaping (T) -> Void) {
        self.closure = closure
    }
    
    func set(_ value: T) {
        closure(value)
    }
}

struct GetterAndSetter<T> {
    private let getter: Getter<T>
    private let setter: Setter<T>

    init(getter: Getter<T>, setter: Setter<T>) {
        self.getter = getter
        self.setter = setter
    }
    
    var value: T {
        get { getter.value }
        set { setter.set(newValue) }
    }
}
```

Reply:
I'm not sure I understand how this would be used + what are the benefits over having the `token` internal

Reply:
The benefit is that we can keep `token` private and send `GetterAndSetter<Token?>` that wraps `client.token` to current-user controller.

Reply:
I see where you're going with this but if we make the setter accessible internally, what difference does it make from having the `token` accessible internally?